sql/internal/sqlx/migrate.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package sqlx
+
+import (
+	"context"
+
+	"ariga.io/atlas/sql/migrate"
+	"ariga.io/atlas/sql/migrate/ent"
+	"ariga.io/atlas/sql/migrate/ent/revision"
+	"entgo.io/ent/dialect/sql"
+)
+
+// A RevisionStorage provides implementation for the migrate.RevisionReadWriter interface.
+type RevisionStorage struct {
```

Comment:
```suggestion
type EntRevision struct {
```

sounds better IMO for an ent-based implementation.